crates/lib3h/src/dht/mirror_dht.rs

```diff
@@ -301,18 +282,46 @@ impl MirrorDht {
             peer_address_list: self.get_other_peer_list(),
             bundle: buf,
         };
-        DhtEvent::GossipTo(gossip_evt)
+        DhtRequestToParent::GossipTo(gossip_evt)
     }
+}
+
+/// Impl DhtActor interface
+impl
+    GhostActor<
+        DhtRequestToParent,
+        DhtRequestToParentResponse,
+        DhtRequestToChild,
+        DhtRequestToChildResponse,
+        Lib3hError,
+    > for MirrorDht
+{
+    fn take_parent_endpoint(&mut self) -> Option<DhtEndpoint> {
+        std::mem::replace(&mut self.endpoint_parent, None)
+    }
+
+    fn process_concrete(&mut self) -> GhostResult<WorkWasDone> {
+        detach_run!(&mut self.endpoint_self, |es| es.process(&mut ()))?;
+        for request in self.endpoint_self.as_mut().drain_messages() {
+            //debug!("@MirrorDht@ serving request: {:?}", request);
+            self.handle_request_from_parent(request)
+                .expect("no ghost errors");
```

Comment:
I wonder if @neonphog has an example of handling this with a result combinator. I think using the [collect()](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect) trick would be appropriate here.